pkg/boot/flags.go

```diff
@@ -54,8 +55,49 @@ func InjectAppWebSocketFlags(command *cobra.Command, hideEngineSpecificAppComple
 	if !hideEngineSpecificAppCompletion {
 		globalFlags.SetAnnotation("app", cobra.BashCompCustom, []string{"__corectl_get_apps"})
 	}
+	command.RegisterFlagCompletionFunc("app", ListValidAppsForCompletion)
 	// Add all global flags to the set of valid config properties.
 	globalFlags.VisitAll(func(flag *pflag.Flag) {
 		dynconf.AddValidConfigFilePropertyName(flag.Name)
 	})
 }
+
+func ListValidAppsForCompletion(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	type ListItem struct {
+		Name string `json:"name"`
+		Id   string `json:"resourceID"`
+	}
+	type ListAppResponse struct {
+		Data []ListItem `json:"data"`
+	}
+	comm := NewCommunicator(cmd)
+	if comm.IsSenseForKubernetes() {
+		restCaller := comm.RestCaller()
+		var result ListAppResponse
+		err := restCaller.CallStd("GET", "v1/items", "", map[string]string{"sort": "-updatedAt", "limit": "30", "name": toComplete}, nil, &result)
+		if err != nil {
+			return []string{}, cobra.ShellCompDirectiveError
+		}
+		completions := make([]string, 0)
+		for _, x := range result.Data {
+			completions = append(completions, x.Name)
+		}
+		return completions, 0
+	} else {
+		ctx, global, _ := comm.OpenGlobalSocket()
+		docList, err := global.GetDocList(ctx)
+
+		if err != nil {
+			return []string{}, cobra.ShellCompDirectiveError
+		}
+		completions := make([]string, 0)
+		for _, x := range docList {
+			if strings.HasPrefix(x.DocName, toComplete) {
+				completions = append(completions, x.DocName)
+			} else if strings.HasPrefix(x.DocId, toComplete) {
+				completions = append(completions, x.DocId)
+			}
+		}
+		return completions, 0
```

Comment:
Should the `ShellCompDirective` be `cobra.ShellCompDirectiveNoFileComp` here as well?